pyiron_base/database/filetable.py

```diff
@@ -56,8 +56,22 @@ class FileTableSingleton(ABCMeta):
     _instances = {}
 
     def __call__(cls, index_from):
-        _path = os.path.abspath(index_from)
+        _path = os.path.abspath(os.path.expanduser(index_from))
         if _path not in cls._instances:
+            path_match_lst = [
```

Comment:
Let's refactor the path matching out as a static method, then we can have a unit test for the behaviour.

Method draft:
```python
@staticmethod
def get_most_precise_common_path(path, reference_paths):
    try:
        return max(
            [
                p for p in reference_paths 
                if os.path.commonpath([path, p]) == p
            ],
            key=len
        )
    except ValueError:
        # When no common path is found
        return None
```

Test sketch:
```python
reference_paths = [
    os.path.abspath(os.path.expanduser("~/work")),
    os.path.abspath(os.path.expanduser("~/work/pyiron"))
]
test_sibling = os.path.abspath(os.path.expanduser("~/personal"))
test_parent = os.path.abspath(os.path.expanduser("~"))
test_identity = os.path.abspath(os.path.expanduser("~/work/pyiron"))
test_child = os.path.abspath(os.path.expanduser("~/work/events"))

for p in [
    test_sibling,
    test_parent,
    test_identity,
    test_child
]:
    print(p, foo(p, reference_paths))
>/Users/huber/personal None
>/Users/huber None
>/Users/huber/work/pyiron /Users/huber/work/pyiron
>/Users/huber/work/events /Users/huber/work
```